third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/TargetInfo.cpp

```diff
@@ -309,6 +309,12 @@ void TargetInfo::storeDShared(RewriterBase &rewriter, Location loc, Value ptr,
   unsigned elemBitwidth = elemTy.getIntOrFloatBitWidth();
   assert(llvm::isPowerOf2_32(vec));
 
+  if (stMatrix) {
+    stMatrixm8n8x4(i32_val(0), unpackLLVector(loc, val, rewriter), 0, ptr,
```

Comment:
Instead of having the additional bool param on storeDShared, should we just have a separate function?

Like, the semantics of storeDShared(stMatrix=false) don't actually seem that similar to storeDShared(stMatrix=true).  For one thing, this doesn't work with distributed shared memory as we've written it (so if we continue using storeDShared we should add an assertion that ctaId is nullopt?).  Also stMatrix=true seems to mean stMatrix8x8x4=true, which is kind of different?  And so on.

Reply:
Agree. It would make the use of `stMatrix` more explicit and solve your questions above.